pkg/cloudarmor/cloudarmor_test.go

```diff
@@ -24,291 +24,398 @@ import (
 			},
-			want: types.False,
 		},
-		{
-			name: "contains success for path",
-			expr: "request.path.lower().contains('bar')",
-			vars: &cloudarmor.Variables{
-				Request: &cloudarmor.Request{
-					Path: "/foo/Bar/baz",
+		want:    types.True,
+		version: cloudarmor.VNext,
+	},
+	{
+		name: "nested params field selection",
+		expr: "request.params.key1.key2.startsWith('value')",
+		vars: &cloudarmor.Variables{
+			Request: &cloudarmor.Request{
+				Params: map[string]any{
+					"key1": map[string]any{
+						"key2": "value2",
+					},
 				},
 			},
-			want: types.True,
 		},
-		{
-			name: "has header",
-			expr: `has(request.headers['user-agent'])`,
-			vars: &cloudarmor.Variables{
-				Request: &cloudarmor.Request{
-					Headers: cloudarmor.HTTPHeaders(map[string]string{"user-agent": "mozilla"}),
+		want:    types.True,
+		version: cloudarmor.VNext,
+	},
+	{
+		name: "nested params backtick selection",
+		expr: "request.params.`key-one`.`key-two`.contains('value')",
+		vars: &cloudarmor.Variables{
+			Request: &cloudarmor.Request{
+				Params: map[string]any{
+					"key-one": map[string]any{
+						"key-two": "value=two",
+					},
 				},
 			},
-			want: types.True,
 		},
-	}
-	rules, err := cloudarmor.NewRules()
-	if err != nil {
-		t.Errorf("cloudarmor.NewRules() returned error: %v", err)
-	}
-	for _, tst := range tests {
-		tc := tst
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
-			ast, iss := rules.Compile(tc.expr)
-			if iss != nil {
-				t.Fatalf("rules.Compile() returned issues: %v", iss)
-			}
-			prg, err := rules.Program(ast)
-			if err != nil {
-				t.Fatalf("ruels.Program() returned error: %v", err)
-			}
-			res, _, err := prg.Eval(tc.vars)
-			if err != nil {
-				t.Errorf("prg.Eval() returned error: %v", err)
-			}
-			if res != tc.want {
-				t.Errorf("prg.Eval() = %v, want %v", res, tc.want)
+		want:    types.True,
+		version: cloudarmor.VNext,
+	},
+}
+
+func TestRules(t *testing.T) {
+	for _, v := range []uint32{cloudarmor.VCurrent, cloudarmor.VNext} {
+		rules, err := cloudarmor.NewRules(cloudarmor.Version(v))
+		if err != nil {
+			t.Errorf("cloudarmor.NewRules() returned error: %v", err)
+		}
+		for _, tst := range tests {
+			tc := tst
+			if v < tc.version {
+				continue
 			}
-		})
+			t.Run(tc.name, func(t *testing.T) {
+				t.Parallel()
+				ast, iss := rules.Compile(tc.expr)
+				if iss != nil {
+					t.Fatalf("rules.Compile() returned issues: %v", iss)
+				}
+				prg, err := rules.Program(ast)
+				if err != nil {
+					t.Fatalf("ruels.Program() returned error: %v", err)
+				}
+				res, _, err := prg.Eval(tc.vars)
+				if err != nil {
+					t.Errorf("prg.Eval() returned error: %v", err)
+				}
+				if res != tc.want {
+					t.Errorf("prg.Eval() = %v, want %v", res, tc.want)
+				}
+			})
+		}
 	}
 }
 
 func TestRunTestSuite(t *testing.T) {
-	tsData, err := os.ReadFile("../../test/http-tests.yaml")
+	tsData, err := os.ReadFile("experimental/users/tswadell/cloudarmor/testdata/http-tests.yaml")
```

Comment:
Please revert this change to make sure that the file is found in the local test directory